### PR TITLE
change filter by endpoint from the store

### PR DESCRIPTION
**Pull request by amiryehu**

### CODE REVIEW

File: src/components/Filter/Filter.tsx [hunk 1]
```diff
@@ -23,14 +21,60 @@ export default function BasicSelect(props: IFilter) {
   const { title, items, isSearchFilter } = props;
   const [selectedItem, setSelectedItem] = useState("");
 
+  useEffect(()=>{
+    if(isSearchFilter){
```
Comment (ofekbd-moveo) [inline src/components/Filter/Filter.tsx:25]: isEndpointFilter would be more clear

File: src/components/Filter/Filter.tsx [hunk 2]
```diff
@@ -23,14 +21,60 @@ export default function BasicSelect(props: IFilter) {
...
 
+  useEffect(()=>{
+    switch (title){
+      case Endpoint.Everything: {
+        store.dispatch(filtersActions.setEndpoint(selectedItem));
+        break;
+      }
+      case Endpoint.TopHeadlines: {
+        store.dispatch(filtersActions.setEndpoint(selectedItem));
+        console.log("title:",title,"selected",selectedItem)
```
Comment (ofekbd-moveo) [inline src/components/Filter/Filter.tsx:46]: delete

File: src/components/Filter/Filter.tsx [hunk 3]
```diff
@@ -23,14 +21,60 @@ export default function BasicSelect(props: IFilter) {
...
+        break;
+      }
+      case COUNTRY: {
+        store.dispatch(filtersActions.setCountry(selectedItem));
+        console.log("title:",title,"selected",selectedItem)
```
Comment (ofekbd-moveo) [inline src/components/Filter/Filter.tsx:51]: delete

File: src/components/MainContent/MainContent.tsx [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
 const Body = () => {
+  const totalResults = useAppSelector(state => state.apiArticlesResponse.totalResults);
+  const displayTotalResults = () => {
+    return `${totalResults} Total results`
+  }
   return (
     <AllBody>
       <HeaderMain>
         <FilterBar />
+        <TotalResults>{displayTotalResults()}</TotalResults>
```
Comment (ofekbd-moveo) [inline src/components/MainContent/MainContent.tsx:17]: <TotalResults>{`${totalResults} Total results}</TotalResults>

File: src/components/CardContainer/CardContainer.tsx [hunk 5]
```diff
@@ -5,22 +5,25 @@ import { ICard } from "../../Utils/Types";
...
+import { EVERYTHING_URL, TOP_HEADLINES_URL } from "../../store/Utils/storeConstances"
 
-const CardContainer = () => {
 
+const CardContainer = () => {
   const dispatch = useAppDispatch();
   
   useEffect(()=>{
-    dispatch(fetchFromApi(EVERYTHING_STRING + API_KEY));
+    dispatch(fetchFromApi(EVERYTHING_URL + API_KEY));
```
Comment (edenmoveohls) [inline src/components/CardContainer/CardContainer.tsx:15]: I think it is better to store all strings in the same file

File: src/components/CardContainer/CardContainer.tsx [hunk 6]
```diff
@@ -5,22 +5,25 @@ import { ICard } from "../../Utils/Types";
...
   },[])
-  
-  const articles = useAppSelector(state => state.apiResponse.articles);
+
+  const endPoint = EVERYTHING_URL;
+  dispatch(() => endPoint);
+
+  const articles = useAppSelector(state => state.apiArticlesResponse.articles);
```
Comment (edenmoveohls) [inline src/components/CardContainer/CardContainer.tsx:21]: use state type 
(state:Check its type)=>state. .... )

File: src/components/Filter/Filter.tsx [hunk 7]
```diff
@@ -23,14 +21,60 @@ export default function BasicSelect(props: IFilter) {
   const { title, items, isSearchFilter } = props;
   const [selectedItem, setSelectedItem] = useState("");
 
+  useEffect(()=>{
+    if(isSearchFilter){
+      setSelectedItem(()=> Endpoint.TopHeadlines);
```
Comment (edenmoveohls) [inline src/components/Filter/Filter.tsx:26]: I think that if you already have redux store, do not use local states, but its yours choise

File: src/components/Filter/Filter.tsx [hunk 8]
```diff
@@ -23,14 +21,60 @@ export default function BasicSelect(props: IFilter) {
...
   };
 
+  useEffect(()=>{
+    switch (title){
+      case Endpoint.Everything: {
+        store.dispatch(filtersActions.setEndpoint(selectedItem));
+        break;
+      }
+      case Endpoint.TopHeadlines: {
+        store.dispatch(filtersActions.setEndpoint(selectedItem));
```
Comment (edenmoveohls) [inline src/components/Filter/Filter.tsx:45]: you can use dispatch(filterActions.setEnpoints)
you dont need to import store

File: src/components/Filter/Filter.tsx (hunk 3 above)
Comment (edenmoveohls) [inline src/components/Filter/Filter.tsx:51]: consloe log is used for debug, removed it after you done

File: src/components/Filter/Filter.tsx [hunk 9]
```diff
@@ -23,14 +21,60 @@ export default function BasicSelect(props: IFilter) {
...
+      }
+      case SOURCES: {
+        store.dispatch(filtersActions.setSources(selectedItem));
+        break;
+      }
+      case SORTBY: {
+        store.dispatch(filtersActions.setSortBy(selectedItem));
+        break;
+      }
+      case DATES: {
```
Comment (edenmoveohls) [inline src/components/Filter/Filter.tsx:66]: switch case it is really expensive function, you can use something more generic

File: src/components/FilterBar/FilterBar.tsx [hunk 10]
```diff
@@ -1,20 +1,35 @@
 import React from "react";
 import { FiltersBar } from "./FilterBar-style";
 import BasicSelect from "../Filter/Filter";
-import BasicDatePicker from "../DatrPicker/DatePicker";
 import ChooseDate from "../DatrPicker/DatePicker";
-
+import {  COUNTRY,  CATEGORY,  SOURCES,  SORTBY,  DATES,  LANGUAGE,  ENDPOINT,  Endpoint,} from "../../store/Utils/storeConstances";
+import { store, useAppSelector } from "../../store/store";
```
Comment (edenmoveohls) [inline src/components/FilterBar/FilterBar.tsx:6]: dont import store

File: src/components/CardContainer/CardContainer.tsx (hunk 5 above)
Comment (amiryehu) [reply src/components/CardContainer/CardContainer.tsx:15]: all strings in the "storeConstances" file.
Is this what you meant?

**Pull request merged**